docs/reference/ssl/kubernetes-with-cert-manager.md

```diff
@@ -0,0 +1,203 @@
+```
+
+```sh
+$ kubectl apply -f letsencrypt-issuer.yaml
+```
+
+This will allow `cert-manager` to automatically provision Certificates just in the `openfaas` namespace.
+
+## Add TLS to openfaas
+
+The OpenFaaS Helm Chart already supports the nginx-ingress, but we want to customize it further. This is easiest with a custom values file. Below, we enable and configure the ingress object to use our certificate and expose just the gateway
+
+```yaml
+# tls.yml
+ingress:
+    enabled: true
+    annotations:
+        kubernetes.io/ingress.class: nginx
+        certmanager.k8s.io/cluster-issuer: letsencrypt-staging
+    ingress:
+        tls:
+        - hosts:
+          - openfaas.mydomain.com
+          secretName: openfaas-crt
+        hosts:
+        - host: openfaas.mydomain.com
+            serviceName: gateway
+            servicePort: 8080
+            path: /
+```
+
+
+```sh
+$ helm upgrade openfaas \
+    --namespace openfaas \
+    ----reuse-values \
+    --values tls.yml \
+    openfaas/openfaas
+```
+
+## Create a certificate
+
+Finally, we can create the Certificate resource which triggers the actual creation of the certificate by `cert-manager`, edit the file below and replace the text `openfaas.mydomain.com` with your address for the API Gateway.
+
+```yaml
+# openfaas-crt.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: openfaas-crt
+spec:
+  secretName: openfaas-crt
+  dnsNames:
+    - openfaas.mydomain.com
+  acme:
+    config:
+      - http01:
+          ingressClass: nginx
+        domains:
+          - openfaas.mydomain.com
+  issuerRef:
+    name: letsencrypt-staging
+    kind: ClusterIssuer
+```
+
+```sh
+$ kubectl apply -f openfaas-crt.yaml
+```
+
+You can validate that certificate has been obtained successfully using
+
+```sh
+$ kubectl describe certificate openfaas-crt
+```
+
+If it was successful you can change to the production Let's Encrypt issuer by replacing `letsencrypt-staging` with `letsencrypt-prod` in the Certificate object.
+
+## Deploy and Invoke a function
+
+In your projects containing OpenFaaS functions, you can now deploy using your domain as the gateway, replace `openfaas.mydomain.com` with your domain as well as adding the username and password you created when you deployed OpenFaaS.
+
+```sh
+faas-cli login --gateway https://openfaas.mydomain.com --username <username> --password <password>
+faas-cli deploy --gateway https://openfaas.mydomain.com
+```
+
+## Verify and Debug
+
+There are several commands we can use to verify that the required kubernetes objects have been created.
+
+- To check that the cert-manager issuers were created:
+```sh
+$ kubectl -n openfaas get issuer letsencrypt-prod letsencrypt-staging
+```
+
+- To check that your certificate was created and that cert-manager created the required secret with the actual ssl certificate:
+```sh
+$ kubectl -n openfaas get certificate,secret openfaas-crt
+```
+
+- To check the access and error logs in Nginx
+```sh
```

Comment:
Let's give 1 instead of 3 commands to get Nginx logs to keep things concise and easier to maintain. Can we just pass `--tail 100 -f` for instance?

The logs / events of the issuer would also be useful. I think this is what I used when we had the issues with cert-manager not refreshing the OFC certs.

Btw ssl should be SSL everywhere right?